### PR TITLE
Updated FAQ and Some Docstrings

### PR DESCRIPTION
**Pull request by shahidee44**
## Description

- Added FAQ to ReadtheDocs Source
- Updated Docstrings for a couple of files.

## Checklist

[//]: <> (- [ ] My code follows the style guidelines of this project)

- [x] I have performed a self-review of my code.
- [x] I have commented my code and used numpy-style docstrings
- [x] I have made corresponding updates to the documentation.
- [x] My changes generate no new warnings
- [ ] I have added/updated tests to make sure bugfix/feature works.
- [x] New and existing unit tests pass locally with my changes.

[//]: <> (- [ ] Any dependent changes have been merged and published in downstream modules)

## Type of change

- [x] Documentation update

## How Has This Been Tested?

test_all.py

### CODE REVIEW

File: docs/source/faq.rst [hunk 1]
```diff
@@ -2,3 +2,62 @@
...
+    QAOA and a discretised adiabatic annealing process. If we pick a linear ramp annealing schedule, i.e. :math:`s(t) = \frac{t}{\tau}`, where :math:`\tau` is the total
+    annealing time, we need to specify two numbers: the total annealing time :math:`\tau` and the step width
+    :math:`\Delta t`. Equivalently, we can also specify the total annealing time :math:`\tau` together with
+    the number of steps :math:`n_{\textrm{steps}}`, which is also called `p` in the
+    context of QAOA. A good discretised annealing schedule has to strike a
+    balance between a long annealing time :math:`\tau` and a small step width
+    :math:`\Delta t = \frac{\tau}{n_{\textrm{steps}}}`. We have found in numerical
+    experiments that :math:`\Delta t = 0.7 = \frac{\tau}{n_{\textrm{steps}}}` strikes a reasonably good balance
+    for many problem classes and instances, at least for the small system sizes one can feasibly simulate.
+    For larger systems or smaller energy gaps, it might be neccesary to choose smaller values of :math:`\Delta t`.
```
Comment (Q-lds) [inline docs/source/faq.rst:33]: neccesary -> necessary

File: docs/source/faq.rst [hunk 2]
```diff
@@ -2,3 +2,62 @@
...
+    The implementation of this method can be found in the subclasses of ``QAOAVariationalBaseParams``.
+
+Discrete sine and cosine transforms for the ``QAOAVariationalFourierParams`` class
```
Comment (Q-lds) [inline docs/source/faq.rst:36]: transforms -> transform

File: docs/source/faq.rst [hunk 3]
```diff
@@ -2,3 +2,62 @@
...
+    :math:`\left|+ \cdots +\right>` is now its minimum energy eigenstate. Conceptually this makes the analogy to adiabatic
+    computing clear, since we seek to transform from the ground state of the mixer Hamiltonian to the ground state of the cost Hamiltonian. 
+
+Implementation of circuit rotation angles
+    In quantum mechanics, the basic time evolution operator is :math:`\exp(-iHt)` for a Hamiltonian `H` and total
+    evolution time `t`. Generically, in the QAOA mixer Hamiltonian, the operator :math:`-X` is to be applied for a total time 
+    :math:`\beta`, which is one of the parameters we seek to optimise. We therefore need to implement the time evolution 
+    :math:`\exp(i\beta X)`, which can be achieved using the RX(:math:`\theta`) operator if we set :math:`\theta = -2\beta`. 
+
+    Similarly, the cost Hamiltonian operator :math:`\exp(-i\gamma hZ)` can be implemented via an RZ(:math:`\theta`) rotation, setting
```
Comment (Q-lds) [inline docs/source/faq.rst:19]: Maybe change the syntax to

You can verify these details in the methods that relate to the creation of these angles in ``openqaoa.qaoa_parameters.standardparams.QAOAVariationalStandardParams``.

File: docs/source/faq.rst (hunk 2 above)
Comment (vishal-ph) [reply docs/source/faq.rst:36]: I believe transforms is used as a noun here (not a verb), i.e. sine transform and cosine transform -> sine and cosine transforms. So the correct word would be "transformations"

File: docs/source/faq.rst (hunk 1 above)
Comment (shahidee44) [reply docs/source/faq.rst]: Resolved in latest commit

File: docs/source/faq.rst (hunk 3 above)
Comment (shahidee44) [reply docs/source/faq.rst:19]: Resolved in latest commit

File: docs/source/faq.rst (hunk 2 above)
Comment (Q-lds) [reply docs/source/faq.rst]: Yes, you are right ^^

Comment (shahidee44) [reply docs/source/faq.rst]: transformations sounds more appropriate. Resolved in latest commit.

**Review by Q-lds — APPROVED** [review]

**Pull request merged**